### PR TITLE
Set object in Cookie

### PR DESCRIPTION
**Pull request by januszhou**
Add ability to set object in Cookie and get object by using `getJSON`

### CODE REVIEW

**Review by alfhen — CHANGES_REQUESTED** [review]
Hey @januszhou, 
Thanks for taking an interest and making this suggestion, I can see where it might be useful. However I am not sure I agree with the implementation. 

For instance you have modified the ```set()``` method to always run a ```JSON.stringify()``` test on the however to get the value you have to call ```getJSON()``` and not just ```get()```, meaning you know before you get the cookie wether a object was stored or not, therefore you should also have this knowledge when saving the cookie. 
Also I do not wish the set function to make any assertions based on the value being passed, its only job should be to take the value and save it in a cookie.

I think it would make more sense to have a ```setObject()``` which just aliases the ```set``` function but with the addition that it runs a ```JSON.stringify()``` on the value before passing it to the ```set``` and then the opposite of that would be ```getObject()```

The implementation of ```setObject()``` could look something like 
``` javascript

setObject: function (name, object, daysOrOptions) {
            try {
              var result = JSON.stringify(value);
              if (/^[\{\[]/.test(result)) {
                value = result;
              }
            } catch (e) {
                return false;
            }

            return this.set(name, value, opts);
        },
```

The implementation of ```getObject()``` could look something like 
``` javascript
getObject: function (name) {
            var cookie = Cookie.get(name);

            try {
              cookie = JSON.parse(cookie);
            } catch (e) {
              return false;
            }

            return cookie;
        },
```

I hope you can see what I mean.

**Comment by januszhou** [comment]
Hi @alfhen, I think you made the point. I made few updates, the only different compare your sample code is I didn't return boolean if exception happened, so even you call `setObject` but value is string, it will still pass. Same applies to `getObject`.

File: src/vue-cookie.js [hunk 1]
```diff
@@ -17,12 +17,31 @@
...
         },
+        setObject: function(name, value, daysOrOptions) {
+            try {
+                var result = JSON.stringify(value);
+                if (/^[\{\[]/.test(result)) {
+                    value = result;
+                }
+            } catch (e) {}
 
+            return this.set(name, value, opts);
```
Comment (alfhen) [inline src/vue-cookie.js]: I see only one problem, here you are passing the third parameter ```opts``` to ```this.set()``` but the third parameter for ```setObject()``` is called ```daysOrOptions``` thus you will always be passing ```undefined``` to ```this.set()```

**Review by alfhen — CHANGES_REQUESTED** [review]
Hi @januszhou, looks good, there is one small typo though that you need to fix, if you do that then I can merge and make a new release.

**Comment by januszhou** [comment]
@alfhen Nice catch, it has been fixed.

Besides that, do you have any idea why circleCI didn't runs my setObject test? 

Test result as below.
```
ok 1 Firefox 47.0 - VueCookie Should set and retrieve a Cookie with given value.
ok 2 Firefox 47.0 - VueCookie Should delete existing cookie and get null when fetching deleted cookie.
ok 3 Firefox 47.0 - VueCookie Should set and retrieve a Cookie with given value from a domain.
ok 4 Firefox 47.0 - VueCookie Should delete existing cookie with a domain and get null when fetching deleted cookie.
ok 5 Chrome 54.0 - VueCookie Should set and retrieve a Cookie with given value.
    ---
        Log: |
            { type: 'log',
              text: '\'Download the Vue Devtools for a better development experience:\\nhttps://github.com/vuejs/vue-devtools\'\n' }
    ...
ok 6 Chrome 54.0 - VueCookie Should delete existing cookie and get null when fetching deleted cookie.
ok 7 Chrome 54.0 - VueCookie Should set and retrieve a Cookie with given value from a domain.
ok 8 Chrome 54.0 - VueCookie Should delete existing cookie with a domain and get null when fetching deleted cookie.
ok 9 PhantomJS 2.1 - VueCookie Should set and retrieve a Cookie with given value.
ok 10 PhantomJS 2.1 - VueCookie Should delete existing cookie and get null when fetching deleted cookie.
ok 11 PhantomJS 2.1 - VueCookie Should set and retrieve a Cookie with given value from a domain.
ok 12 PhantomJS 2.1 - VueCookie Should delete existing cookie with a domain and get null when fetching deleted cookie.
```

**Comment by alfhen** [comment]
@januszhou The reason why CircleCi does not run your test is because it has problems compiling the files on the virtual server it runs, therefore whenever you make changes to the tests, you have to run ```npm run build-test``` from the root of the project. This will compile the spec file, then you must commit it along with the rest of your changes. I now it is troublesome, but I haven't had the time to fix the compiling on CircleCI. It is my wish to replace it with Webpack at some point when I find the time.

**Comment by caiyawen** [comment]
Could you tell me when do you attend to merge this request since I need this feature ?

**Pull request closed without merging**